### PR TITLE
add flux correction for bubble

### PR DESCRIPTION
**Pull request by simonbyrne**
- [x] Confirm that correct (with correct sign)
- [x] Add unit test (e.g. 1D column
@simonbyrne  and @akshaysridhar

### CODE REVIEW

File: src/Operators/finitedifference.jl [hunk 1]
```diff
@@ -1078,6 +1078,177 @@ function stencil_right_boundary(
...
+        getidx(velocity_field, loc, idx + half),
+        Geometry.LocalGeometry(space, idx + half),
+    )
+    w³⁻ = Geometry.contravariant3(
+        getidx(velocity_field, loc, idx - half),
+        Geometry.LocalGeometry(space, idx - half),
+    )
+    ∂θ₃⁺ = θ⁺ ⊟ θ
+    ∂θ₃⁻ = θ ⊟ θ⁻
+    return (abs(w³⁺) ⊠ ∂θ₃⁺) ⊟ (abs(w³⁻) ⊠ ∂θ₃⁻)
```
Comment (simonbyrne) [inline src/Operators/finitedifference.jl:1121]: @akshaysridhar this is the flux correction operator

**Comment by simonbyrne** [comment]
@akshaysridhar can you take this on? I'm happy to help out where need be

**Comment by akshaysridhar** [comment]
all_advect_end Test against sin function transport

all_step_end

Dash => UBP
Dot => UBP + FluxCorrection
Solid => C2C 
DashDot => C2C + FluxCorrection

Tests of operator against sine wave, and step function advection

**Comment by akshaysridhar** [comment]
@simonbyrne Please confirm that the configuration in the `step.jl` addition to the `advect` folder is sensible. I think this is ready to be merged.

**Comment by jakebolewski** [comment]
@akshaysridhar  looks good, you don't strictly need BC's for every operator like that but it does make it easier to deubg sub-setting terms in the RHS update if they are.

**Comment by akshaysridhar** [comment]
bors try

**Comment by akshaysridhar** [comment]
@jakebolewski Nightly 1.7 build tests are failing ; is this expected based on the changes Simon and I have made ? If not, what is the solution to this?

**Pull request merged**